Provider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbTransaction.cs

```diff
@@ -173,92 +145,77 @@ public override void Rollback()
 		public void Save(string savePointName)
 		{
 			EnsureSavePointName(savePointName);
-			lock (this)
+			EnsureCompleted();
+			try
 			{
-				EnsureCompleted();
-				try
-				{
-					using (var command = new FbCommand($"SAVEPOINT {savePointName}", _connection, this))
-					{
-						command.ExecuteNonQuery();
-					}
-				}
-				catch (IscException ex)
+				using (var command = new FbCommand($"SAVEPOINT {savePointName}", _connection, this))
 				{
-					throw new FbException(ex.Message, ex);
+					command.ExecuteNonQuery();
 				}
 			}
+			catch (IscException ex)
+			{
+				throw new FbException(ex.Message, ex);
+			}
 		}
 
 		public void Commit(string savePointName)
 		{
 			EnsureSavePointName(savePointName);
-			lock (this)
```

Comment:
Yeah. It's removed everywhere. It was part of the motivation to fix the finalizers mess.